docs/content/topics/scripting/guide.md

```diff
@@ -9,1401 +9,867 @@ aliases:
-
-The `after` hook runs _at the very end_ of any session that did not die from an error.
-
-```javascript
-const { events } = require("brigadier")
-
-events.on("exec", () => {
-  console.log("==> handling an 'exec' event")
-})
-
-events.on("after", () => {
-  console.log(" **** AFTER EVENT called")
-})
+Waiting for event's worker to be RUNNING...
+2021-09-20T22:29:28.498Z INFO: brigade-worker version: a398ba8-dirty
+Hello, World!
 ```
-[brigade-04.js](../../examples/brigade-04.js)
 
-The `brig` client will trigger the `exec` event. But then when that event has been
-processed, Brigade will trigger an `after` event before returning:
+Since Brigade did not see a GitHub push event (in this case, an event from
+source `brigade.sh/github` and of type `push`), it did not execute the
+corresponding event handler. It only executed the handler for the event created
+by `brig`, i.e. an event from source `brigade.sh/cli` and of type `exec`.
 
-```console
-Started brigade-worker-01brx76gx5v3e8r6vbmzcda7q9-master
-yarn start v0.27.5
-$ node prestart.js
-prestart: src/brigade.js written
-$ node --no-deprecation ./dist/src/index.js
-Creating PVC named brigade-worker-01brx76gx5v3e8r6vbmzcda7q9-master
-==> handling an 'exec' event
- **** AFTER EVENT called
-Destroying PVC named brigade-worker-01brx76gx5v3e8r6vbmzcda7q9-master
-Done in 1.19s.
-```
-
-Note that the `**** AFTER EVENT called` message is printed right before the shutdown
-begins.
+See the [Events] doc for more info on how events are structured in Brigade.
 
-The `error` event is similar, but it is _only_ triggered when a script encounters an
-error.
-
-The `after` and `error` events give you an opportunity to do some final processing (like
-maybe sending a message to your chat server) before exiting.
-
-At this point, we've taken a look at Brigade's event system. But we haven't really done
-anything with Brigade except log a few messages. Let's turn out attention toward Brigade's
-ability to "script up" containers.
+[Events]: /topics/project-developers/events
+[event summary]: /topics/project-developers/events#summary
 
 ## Jobs and Containers
 
-A typical Brigade script divides up the workload like this:
+A typical Brigade script divides up the workload for handling events like this:
 
-- An _event handler_ declares how an event (`push`, `exec`) is processed
-- When Brigade triggers an event, it creates a new `build`, which you can think of as "a
-  specific run of your `brigade.js` file.
-- A build has several _jobs_, where each job starts a container
-- A job runs zero or more _tasks_ inside of a container
+- An _event handler_ declares how an event (`brigade.sh/github:push`,
+  `brigade.sh/cli:exec`) is processed
+- When Brigade triggers an event that a project subscribes to, Brigade creates
+  a _Worker_ per the project's configuration
+- A Worker may run several _Jobs_, where each job starts a container
+- A Job runs zero or more _commands_ inside of a container
```

Comment:
Two things:

Can we simplify much of this by saying that Brigadier exposes an API for defining and executing jobs?

Can we explain the benefit of jobs somehow?

A digression we don't necessary need to include-- in v1, we seemed to discourage people from performing actual work in their workers and we aren't really discouraging that now, which I think is good, but... `</digression>`

The worker's image is derived from a Node.js image and containers based on that image can execute JavaScript and TypeScript only. In the frequent case that other runtimes, tools, etc. _not_ present on the worker image are required in order to respond to an event, spawning a job permits aspects of event-handling to be delegated to a more suitable container.